src/renderers/shaders/UniformsUtils.js

```diff
@@ -14,14 +14,11 @@ export function cloneUniforms( src ) {
 
 			const property = src[ u ][ p ];
 
-			if ( property && ( property.isColor ||
-				property.isMatrix3 || property.isMatrix4 ||
-				property.isVector2 || property.isVector3 || property.isVector4 ||
-				property.isTexture || property.isQuaternion ) ) {
+			if ( property && property.clone !== undefined ) {
```

Comment:
I really don't hope we apply this pattern. This kind of "type check" is just awful...

Reply:
Indeed...

This would allow someone to pass by mistake any object that has a `clone()` (ie. `Object3D`, `Mesh`, `Material`, ...) and things would get very ugly.

Reply:
Yes. This pattern can break in some many ways...and it's not readable at all. 

It does not become clear what type of objects `cloneUniforms()` actually supports. It's definitely not "all objects with a clone() method".

Reply:
I think the method will break in any case if `Object3D`/`Mesh`/etc would be passed to it?

Reply:
Also, why is it awful and not readable?

It is the simplest that it is:
* if value has `clone` method - use it,
* if value has `slice` method - use it,
* otherwise just use the value.

Reply:
Checking for explicit types is more readable and easier to understand than what's happening in this PR. 

Sorry, but this goes in the wrong direction. Let's focus on https://github.com/mrdoob/three.js/pull/24202#issuecomment-1149547425.

Reply:
_Why_ we need there a check for explicit types? Why isn't it better to use duck typing here?

Reply:
Awful sure is a strong word in this case 😅

When we talk about readability we mean it from the point of view of maintainability.
The explicit check makes much more clear what  do we expect to see in that part of the code so if we need to do improvements we (or others) don't spend time trying to relearn the logic.

Reply:
OK, then... Can it just go as a comment like `// The possible uniforms can be numbers, booleans, arrays, vectors, matrices, quaternions, colors, and texture`?
It would be even more readable and will allow to make this change without reducing the readability.

Reply:
I think for now it'll be better to just try the `instanceof` change.